AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/diag-provider.ts

```diff
@@ -140,10 +139,39 @@ export class DiagProvider {
         });
         return Promise.race([promise, timeoutPromise]);
     }
+    public getDaaSExtApiAsync(api: string, params = [], timeoutInSec: number = 15): Promise<any> {      
+        var prefix = `management.azure.com/${this._siteInfo.resourceUri}/extensions/DaaS/api`;
+        var stack = new Error("error_message_placeholder").stack;
+        var promise = this._armService.get(`https://${prefix}/${api}?${params.join("&")}`)         
+            .toPromise()
+            .catch(e => {
+                e.stack = stack.replace("error_message_placeholder", e);
+                throw e;
+            });     
+            var timeoutPromise = delay(timeoutInSec).then(() => {
+                throw new Error(`getDaaSExtApiAsync timeout after ${timeoutInSec}s`);
+            });
+            return Promise.race([promise, timeoutPromise]);        
+    }   
 
     public async checkConnectionStringAsync(connectionString: string, type: string, timeoutInSec: number = 30): Promise<any> {
         var result: any = await this.postDaaSExtApiAsync("connectionstringvalidation/validate", { "ConnectionString": connectionString, "Type": type }, timeoutInSec);
-        return result;
+        return result;       
+    }
+    public async checkConnectionViaAppSetting(appSetting: string, type: string, entityName?: string, timeoutInSec: number = 30): Promise<any> {
+        var param = [];
+        param.push("api-version=2015-08-01");
```

Comment:
Why don't you set the api version in getDaaSExtApiAsync() to be consistent with how we do post?